packages/expo/scripts/aggregate_xcprivacy/aggregate_xcprivacy.rb

```diff
@@ -0,0 +1,91 @@
+require_relative './plist/lib/plist'
+
+def get_used_required_reason_apis(installer)
+  # A dictionary with keys of type string (NSPrivacyAccessedAPIType) and values of type string[] (NSPrivacyAccessedAPITypeReasons[])
+  used_apis = {}
+  puts "Reading xcprivacy files to aggregate all used Required Reason APIs."
+  installer.pod_targets.each do |pod_target|
+    pod_target.file_accessors.each do |file_accessor|
+      file_accessor.resource_bundles.each do |bundle_name, bundle_files|
+        bundle_files.each do |file_path|
+          # This needs to be named like that due to apple requirements
+          if File.basename(file_path) == 'PrivacyInfo.xcprivacy'
+            content = Plist.parse_xml(file_path)
+            accessed_api_types = content["NSPrivacyAccessedAPITypes"]
+            accessed_api_types.each do |accessed_api|
+              api_type = accessed_api["NSPrivacyAccessedAPIType"]
+              reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+              used_apis[api_type] ||= []
+              used_apis[api_type] += reasons
+            end
+          end
+        end
+      end
+    end
+  end
+  return used_apis
+end
+
+def get_privacyinfo_file_path(target)
+  # We find a file we know exists in the project to get the path to the main group directory
+  app_delegate_path = target.source_build_phase.files_references.find { |file_ref| file_ref.name == "AppDelegate.mm" }
```

Comment:
that's tricky. For brownfield app and custom apps, the `AppDelegate.mm` might be called in different ways.
A check like `file_ref.name.end_with? "AppDelegate.mm"` might be better but still brittle.

However, React Native has the path to the app as a parameter. And I think Cocoapods has a way to retrieve the path of the `Xcodeproj`. We might use that information to find the proper path.

Reply:
I'll assume 1, since it should not cause any issues even if this folder doesn't exist, which should not be the case for almost all projects.